### PR TITLE
fix(structured data): notion DB name should not be empty

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

[datadog](https://app.datadoghq.eu/logs?query=unhandled%20&cols=host%2Cservice&event=AgAAAY72_u-WzdOZdQAAAAAAAAAYAAAAAEFZNzJfdm9kQUFDdmhQOE9aRzdPSkFDbgAAACQAAAAAMDE4ZWY2ZmYtMWZkYy00ZTQ2LTlmYjgtNzZkYzQ0YmQyMjE3&fromUser=true&index=%2A&messageDisplay=inline&refresh_mode=sliding&storage=hot&stream_sort=desc&viz=stream&from_ts=1713532682093&to_ts=1713533582093&live=true)

Notion db title may be an empty string. We don't allow it.

### CODE REVIEW

File: connectors/src/connectors/notion/temporal/activities.ts [hunk 1]
```diff
@@ -2442,10 +2442,18 @@ function getTableInfoFromDatabase(database: NotionDatabase): {
   tableDescription: string;
 } {
   const tableId = getNotionDatabaseTableId(database.notionDatabaseId);
-  const name =
-    database.title ?? `Untitled Database (${database.notionDatabaseId})`;
-  const tableName = slugify(name.substring(0, 32));
+  const fallbackName = `Untitled Database (${database.notionDatabaseId})`;
+
+  let tableName = slugify((database.title ?? "").substring(0, 32));
+  if (!tableName) {
```
Comment (flvndvd) [inline connectors/src/connectors/notion/temporal/activities.ts:2448]: `slugify` always returns a string, so worst case `tableName` will be an empty string, no?

Comment (flvndvd) [reply connectors/src/connectors/notion/temporal/activities.ts:2448]: ok but then why not:
```
let tableName = slugify((database.title ||  fallbackName).substring(0, 32));
```

Comment (fontanierh) [reply connectors/src/connectors/notion/temporal/activities.ts:2448]: Because you have no guarantee that slugify does not replace some characters by `""`

Comment (fontanierh) [reply connectors/src/connectors/notion/temporal/activities.ts:2448]: Today is not the case but in the future you don't know

**Pull request merged**